includes/amp-helper-functions.php

```diff
@@ -1560,10 +1560,12 @@ function amp_get_schemaorg_metadata() {
 		 *
 		 * @since 0.3
 		 *
-		 * @param array   $metadata Metadata.
-		 * @param WP_Post $post     Post.
+		 * @param array   $metadata       Metadata.
+		 * @param WP_Post $queried_object Post.
 		 */
-		$metadata = apply_filters( 'amp_post_template_metadata', $metadata, $post );
+		$metadata = apply_filters( 'amp_post_template_metadata', $metadata, $queried_object );
+	} elseif ( is_archive() ) {
+		$metadata['@type'] = 'collection';
```

Comment:
I agree with @pierlon, we should use `CollectionPage` here, as it is a `WebPage` _containing_  a `Collection`.

Also, I'd stick to the casing of the official type reference.
```suggestion
		$metadata['@type'] = 'CollectionPage';
```